solr/core/src/java/org/apache/solr/cloud/api/collections/DeleteCollectionCmd.java

```diff
@@ -92,6 +98,19 @@ public void call(ClusterState state, ZkNodeProps message, @SuppressWarnings({"ra
       collection = extCollection;
     }
 
+    PlacementPlugin placementPlugin = ocmh.overseer.getCoreContainer().getPlacementPluginFactory().createPluginInstance();
```

Comment:
Didn't dig into the details here, but when we delete a collection, we should just check if there's another collection that defines `withCollection` on it and refuse the delete based on that, no?
Just starting to look at the PR so maybe there's a reason for doing it this way (in which case maybe adding a comment?)

Reply:
The check can be simplified in the plugin if it only knew that we're deleting the whole collection. I guess we could create a `DeleteCollectionRequest` to let it know the type of modification we want to do.